docs/tests.md

```diff
@@ -0,0 +1,439 @@
+`maxUnhelthy` test first which does not suppose machine remediation. Test for actual remediation might be done on the same machineset. 
+
+#### MachineHealthCheck should remediate unhealthy nodes
+* Parallel: Yes
+* ExecTime: >6m on AWS
+
+Within this test, several machines (for now `maxUnhelthy` - 1, which is currently 1) marking as unhealthy with a specific condition.
+Then MHC creating, such MHC is configured to observe this condition.
+After this test waits for unhealthy machines to be deleted and checks that healthy machines stay intact.
+
+This test exercises MHC remediation functionality.
+
+#### MachineHealthCheck should not remediate larger number of unhealthy machines then maxUnhealthy
+* Parallel: Yes
+* ExecTime: >9m on AWS
+
+Within this test, a number of machines exceeding `maxUnhelthy` marking as unhealthy with a specific condition.
+Then MHC creating, such MHC is configured to observe this condition.
+Test waits for a `RemediationRestricted` event on the machineset and checks that no machines were deleted.
+
+This test exercises MHC behaviour that prevents remediation if a number of machines marked as unhealthy more than a number in `maxUnhelthy` parameter.
+
+### Autoscaler
+
+#### Autoscaler should use a ClusterAutoscaler that has 100 maximum total nodes count It scales from/to zero
+* Parallel: No
+* ExecTime: >7m on AWS
+* Recommendation:
+  * amount of desired machines might be reduced from 3 to 2
+
+Within this test machineset with the initial desired 0 replicas and with a specific set of labels is created.
+After, a Job with 3 desired replicas is attempting to be scheduled on nodes with labels configured for previously created machineset.
+
+The test is expecting machineset to be scaled up to 3 machines and accept the payload. Then, created Job is removing and the machineset is expected to be scaled down back to 0 replicas. 
+
+Test considered failed if scale up or down did not happen within 15 min timeout.
+
+#### Autoscaler should use a ClusterAutoscaler that has 100 maximum total nodes count cleanup deletion information after scale down [Slow]
+* Parallel: No
+* ExecTime: ~13m on AWS
+* Recommendation:
+  * This test is a bit cryptic at first glance and requires at least some in-code explanation about the meaning of checks for taints and annotations presence. Having some documentation references will be beneficial here.
+  * amount of desired machines might be reduced from 3 to 2 for each machineset
+
+Within this test machineset with 1 desired replica and with a specific set of labels is created.
+After, a Job with 6 desired replicas is attempting to be scheduled on nodes with labels configured for previously created machinesets.
+
+The test is expecting both machinesets to be scaled up to 3 machines and accept the payload. Then, created Job is removing and the machineset is expected to be scaled down back to 1 replica.
```

Comment:
This seems like it has a big overlap with the previous test right? Do we want to consolidate the two of them/maybe remove the first? Is the first covering something that isn't covered in this test?

Reply:
First one testing scaling from/to zero. This one checks that machine which stays after scaling down is not polluted with unwanted labels/annotations.

But it might be combined with `MaxNodesTotal` test.

Reply:
> But it might be combined with MaxNodesTotal test.

+1